kahypar/partition/bin_packing/bin_packer.h

```diff
@@ -0,0 +1,124 @@
+/*******************************************************************************
+ * This file is part of KaHyPar.
+ *
+ * Copyright (C) 2020 Nikolai Maas <nikolai.maas@student.kit.edu>
+ *
+ * KaHyPar is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * KaHyPar is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with KaHyPar.  If not, see <http://www.gnu.org/licenses/>.
+ *
+******************************************************************************/
+
+#pragma once
+
+#include "kahypar/definitions.h"
+#include "kahypar/partition/bin_packing/i_bin_packer.h"
+#include "kahypar/partition/bin_packing/bin_packing_utils.h"
+#include "kahypar/partition/bin_packing/bin_packing_algorithms.h"
+#include "kahypar/partition/bin_packing/prepacking.h"
+
+namespace kahypar {
+namespace bin_packing {
+// Implementation of IBinPacker for a generic bin packing algorithm.
+template< class BPAlgorithm = WorstFit >
+class BinPacker final : public IBinPacker {
+ public:
+  BinPacker(Hypergraph& hypergraph, Context& context) : _hypergraph(hypergraph), _context(context) {}
+
+ private:
+  void prepackingImpl(const BalancingLevel level) override {
+    ASSERT(!_hypergraph.containsFixedVertices(), "No fixed vertices allowed before prepacking.");
+    ASSERT(level != BalancingLevel::STOP, "Invalid balancing level: STOP");
+
+    const PartitionID rb_range_k = _context.partition.rb_upper_k - _context.partition.rb_lower_k + 1;
+    const HypernodeWeight max_bin_weight = floor(_context.initial_partitioning.current_max_bin_weight * (1.0 + _context.initial_partitioning.bin_epsilon));
+
+    if (level == BalancingLevel::heuristic) {
+      bin_packing::calculateHeuristicPrepacking<BPAlgorithm>(_hypergraph, _context, rb_range_k, max_bin_weight);
```

Comment:
`bin_packing::` shouldn't be necessary here, since you are already in the `bin_packing` namespace.